### PR TITLE
Nohau help guru81

### PR DESCRIPTION
**Pull request by jvela018**
Creating a new branch and pull request to test the Curl demo

### CODE REVIEW

**Comment by bagder** [comment]
1. Please use the description to explain what you PR does. It seems to add a CI job?
2. Please rebase and squash the commits to just the few logical ones you need

File: .github/workflows/codesonar.yaml [hunk 1]
```diff
@@ -0,0 +1,68 @@
+name: CodeSonar
+on:
+  push:                  # Only run onces the Pull request is created
+    branches:
+      - 'release/**'               # Feature Branches
+  pull_request:
+    branches:
+      - 'release/**'
```
Comment (bagder) [inline .github/workflows/codesonar.yaml:8]: We don't use such branches. This looks like bad copy and paste from somewhere else.

File: .github/workflows/codesonar.yaml [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+  push:                  # Only run onces the Pull request is created
+    branches:
+      - 'release/**'               # Feature Branches
+  pull_request:
+    branches:
+      - 'release/**'       
+  
+jobs:
+  CodeSonar_Analyze:
+    permissions: write-all
```
Comment (bagder) [inline .github/workflows/codesonar.yaml:12]: Why does it need write all permission?

File: .github/workflows/codesonar.yaml [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+      ROOT_TREE: "OSS-Projects/curl-GitHub" 
+      PROJECT_NAME: "curl"
+      BRANCH_NAME: ${{ github.head_ref || github.ref_name }} 
+      CSONAR_HUB_URL: "https://duck.dyndns.org:7340" 
+      CSONAR_HUB_USER: "${{ secrets.CSONAR_HUB_USER }}"
+      CSONAR_HUB_PASSWORD: "${{ secrets.CSONAR_HUB_PASS }}"
+      CSONAR_CSHOME: /opt/codesonar
+      CS_USER_HOME: /__w/curl/curl
+      COMMIT_HASH: ${{ github.sha }}
+      REPO_URL: "https://github.com/CodeSecure-SE/curl"    # the github.repositoryUrl is not an http-link.
```
Comment (bagder) [inline .github/workflows/codesonar.yaml:33]: What is this repo?

File: .gitignore [hunk 4]
```diff
@@ -65,3 +65,10 @@ curl_fuzzer_seed_corpus.zip
 libstandaloneengine.a
 tests/string
 tests/config
+.vscode/settings.json
+
+curl.prj_files/*
+curl.conf
+curl.gcc*
+curl.prj
+*.sarif
```
Comment (bagder) [inline .gitignore:74]: I think you should add these to your general git config file and not curl's.

**Review by bagder — CHANGES_REQUESTED** [review]

**Comment by bagder** [comment]
Closing now since there is no point in this wasting CI resources.

**Comment by jvela018** [comment]
@bagder I was supposed to check that I can run the CI against our fork.  Sorry about that.

**Pull request closed without merging**